src/index.ts

```diff
@@ -4,20 +4,31 @@ import * as sass from 'sass';
 import {dirname} from 'path';
 import * as fs from 'fs';
 import {createFilter} from '@rollup/pluginutils';
-import {
+import type {
   SassImporterResult,
   RollupAssetInfo,
   RollupChunkInfo,
   RollupPluginSassOptions,
   RollupPluginSassOutputFn,
-  SassOptions
+  SassOptions,
+  RollupPluginSassProcessorFnOutput,
+  SassRenderResult
 } from "./types";
 import {isFunction, isObject, isString, warn} from "./utils";
 
 // @note Rollup is added as a "devDependency" so no actual symbols should be imported.
 //  Interfaces and non-concrete types are ok.
 import {Plugin as RollupPlugin} from 'rollup';
 
+type PluginState = {
```

Comment:
`PluginState` is required to properly type `processRenderResponse` function which is declared before `pluginState`